Member dropdown

eldiegod opened the pull request:
Fixes: #1645

Testing path: '/playground/member-dropdown'

Figma: https://www.figma.com/file/Pf31tuYpozYmpq163U2ho8/Web-Components?node-id=110%3A18281

Missing things to come later at some point:
- Real Tjoy balance
- Learn about tJOY link
- my profile link
- switch member animation

eldiegod committed:
draft

eldiegod committed:
wip

eldiegod committed:
tjoy dummy

drillprop commented on src/components/_overlays/MemberDropdown/MemberDropdown.tsx:
```diff
@@ -0,0 +1,204 @@
+        <div>
+          <SwitchMemberItemListContainer>
+            <ListItem
+              onClick={() => setIsSwitchingMember(false)}
```
We had already `handleSwitchMemberMode` which we could use here as well(you'd have to add a parameter to the function of course)

eldiegod replied:
I think since the func is just the setter, it's not needed, will remove and use the setter instead for now

drillprop commented on src/components/_overlays/MemberDropdown/MemberDropdown.tsx:
```diff
@@ -0,0 +1,204 @@
+            <ListItem
+              nodeStart={<IconWrapper icon={<SvgActionNewChannel />} />}
+              onClick={() => handleAddNewMember()}
+              label={'Add new member...'}
```
```suggestion
              label="Add new member..."
```
Ok, we should have some eslint/prettier plugin already for changing that 😅  I think [this one](https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/jsx-curly-brace-presence.md) should work, but let's do this in another PR.

drillprop commented on src/components/_overlays/MemberDropdown/MemberDropdown.tsx:
```diff
@@ -0,0 +1,204 @@
+          <BlurredBG url={activeMembership?.avatarUri}>
+            <Filter />
+            <MemberInfoContainer>
+              <StyledAvatar size="fill" assetUrl={activeMembership?.avatarUri}></StyledAvatar>
```
Let's collapse this empty tag

drillprop commented on src/components/_overlays/MemberDropdown/MemberDropdown.tsx:
```diff
@@ -0,0 +1,204 @@
+                    rel="noopener noreferrer"
+                    secondary
+                    color={cVar('colorCoreNeutral200Lighten')}
+                  >
```
In the design `Learn about tJOY` isn't underlined. 
![image](https://user-images.githubusercontent.com/51168865/146331489-d604e4ce-d792-4135-bd3f-83638bb5ab65.png)
But to be honest I like underlined text here. It's suggesting that it's a clickable link. Could you ask @toiletgranny if we could leave this without change as you have here?

drillprop replied:
btw. I'm fine with @ts-ignore

eldiegod replied:
@toiletgranny what do you think?

drillprop replied:
Talked to Adam about this, and he decided that it should be underlined only when you're hovering over this link.

drillprop commented on src/components/_overlays/MemberDropdown/MemberDropdown.tsx:
```diff
@@ -0,0 +1,204 @@
+          </SectionContainer>
+        </div>
+      ) : (
+        <div>
```
I think we don't need a div here, we could use Fragment.

drillprop reviewed (changes requested):
https://user-images.githubusercontent.com/51168865/146333258-022665a1-8bc4-412d-a9f5-656cb24d9f16.mov

I've noticed some layout shifts when switching members. Could we do something about it?
Besides that, found some minor stuff. Looks pretty good in general 👍🏻

kdembler commented:
Is there any specific reason to not use it in Studio in place of the old one right away?

eldiegod committed:
Member dropdown in studio

eldiegod commented:
> Is there any specific reason to not use it in Studio in place of the old one right away?

@kdembler added it to studio topbar but now we are lacking the logout action

eldiegod committed:
Fix bug when switching from member with channel

to member with no channel

eldiegod committed:
playground has no publisher version

eldiegod committed:
fix playground

toiletgranny commented:
@eldiegod, @kdembler, @drillprop Great work, looks stunning! Can't wait to see that transition animation when going into "Switch member" mode! For all other comments and questions, I recorded a quick loom (4 min). https://www.loom.com/share/435e049ab7ce476e8510c2051af73578

eldiegod committed:
cr

eldiegod committed:
box shadow

eldiegod commented:
@toiletgranny I think I implemented everything you mentioned besides the `Select first channel when switching members to a member with channels` part because this is probably more suitable as an issue on its own since it has more to do with our internal user logic than with the member dropdown itself

drillprop commented:
> besides the `Select first channel when switching members to a member with channels` part because this is probably more suitable as an issue on its own since it has more to do with our internal user logic than with the member dropdown itself 

I'll deal with it in https://github.com/Joystream/atlas/pull/1788 or in follow-up PR once this member dropdown PR is merged. I also think that it will be better to merge your commits into `Joystream:signin`, not `Joystream:dev`, wdyt?

drillprop commented on src/components/_navigation/TopbarStudio/TopbarStudio.tsx:
```diff
@@ -50,90 +38,53 @@ type ChannelInfoProps = {
+  //   } else {
+  //     logout()
+  //   }
+  // }
```
Let's just remove this

drillprop commented on src/components/ListItem/ListItem.tsx:
```diff
@@ -40,16 +45,46 @@ export const ListItem = React.forwardRef<HTMLDivElement, ListItemProps>(
+        disabled={disabled}
+        hasNodeStart={!!nodeStart}
+        size={size}
+        ref={mergeRefs([hoverRef, ref])}
```
hmm, that's a cool library!

eldiegod replied:
yea right

toiletgranny commented:
> @toiletgranny I think I implemented everything you mentioned besides the `Select first channel when switching members to a member with channels` part because this is probably more suitable as an issue on its own since it has more to do with our internal user logic than with the member dropdown itself

Ok, thanks! Visually, everything looks good now! Based on your reply, I'd assume that closing the member dropdown upon selecting a different member, which I also mentioned in the loom video, would also be tackled in that separate issue, right?
![CleanShot 2021-12-20 at 12 25 55](https://user-images.githubusercontent.com/35307309/146760389-7d61b55a-d483-4aec-a335-f1800719a05f.gif)

Other than this — it's a go from my side!

eldiegod committed:
close dropdown on member change

eldiegod commented:
> > besides the `Select first channel when switching members to a member with channels` part because this is probably more suitable as an issue on its own since it has more to do with our internal user logic than with the member dropdown itself
> 
> I'll deal with it in #1788 or in follow-up PR once this member dropdown PR is merged. I also think that it will be better to merge your commits into `Joystream:signin`, not `Joystream:dev`, wdyt?

Nice!

~~About merging; I'm not sure, I would merge to dev to not make that release bigger but is not clear, thoughts @kdembler ?~~

nvm we are merging to signin

eldiegod commented:
> > @toiletgranny I think I implemented everything you mentioned besides the `Select first channel when switching members to a member with channels` part because this is probably more suitable as an issue on its own since it has more to do with our internal user logic than with the member dropdown itself
> 
> Ok, thanks! Visually, everything looks good now! Based on your reply, I'd assume that closing the member dropdown upon selecting a different member, which I also mentioned in the loom video, would also be tackled in that separate issue, right? ![CleanShot 2021-12-20 at 12 25 55](https://user-images.githubusercontent.com/35307309/146760389-7d61b55a-d483-4aec-a335-f1800719a05f.gif)
> 
> Other than this — it's a go from my side!

@toiletgranny  No we don't have to wait for that I just added it so we should be feature complete here

kdembler commented on src/components/_overlays/MemberDropdown/MemberDropdown.styles.ts:
```diff
@@ -0,0 +1,113 @@
+  right: ${sizes(4)};
+  top: 0;
+  width: 280px;
+  max-height: calc(${window.innerHeight - sizes(4, true)}px - var(--size-topbar-height));
```
Couldn't we use just `100vh` instead? We have a guarantee this will always be responsive, `window.innerHeight` is iffy

eldiegod replied:
yeah this is better just pushed it

kdembler reviewed (commented):
Just a small comment from myself, other than that looks great to me!

I will still wait for final approval from @drillprop

drillprop reviewed (approved):
Looks good to me! 🎉

eldiegod committed:
better responsiveness
